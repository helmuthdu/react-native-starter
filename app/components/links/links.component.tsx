import * as Samples from '@expo/samples';
import { Body, Button, Container, Content, Header, Icon, Left, Right, Title } from 'native-base';
import React from 'react';
import { ScrollView } from 'react-native';
import { NavigationRoute, NavigationScreenProp } from 'react-navigation';
import { styles } from './links.style';

export const Links = (props: { navigation: NavigationScreenProp<NavigationRoute> }) => (
  <Container style={styles.container}>
    {props.navigation.openDrawer && (
      <Header>
        <Left>
          <Button transparent>
            <Icon active name="menu" onPress={() => props.navigation.openDrawer()} />
          </Button>
        </Left>
        <Body>
          <Title>Links</Title>
        </Body>
        <Right />
      </Header>
    )}
    <Content>
      <ScrollView style={styles.container}>
        {/* Go ahead and delete ExpoLinksView and replace it with your
         * content, we just wanted to provide you with some helpful links */}
        <Samples.ExpoLinksView />
      </ScrollView>
    </Content>
  </Container>
);
