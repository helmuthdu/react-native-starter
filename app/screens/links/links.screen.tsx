import React, { Component } from 'react';
import { Links } from '../../components/links/links.component';

type Props = {
  navigation: any;
};

export default class LinksScreen extends Component<Props> {
  public render() {
    return <Links navigation={this.props.navigation} />;
  }
}
