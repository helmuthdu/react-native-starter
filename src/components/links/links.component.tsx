import React from 'react';
import * as Samples from '@expo/samples';
import { NavigationRoute, NavigationScreenProp } from 'react-navigation';
import { Body, Button, Container, Content, Header, Icon, Left, Right, Title } from 'native-base';
import { styles } from './links.style';
import { ScrollView } from 'react-native';

export const Links = (props: { navigation: NavigationScreenProp<NavigationRoute> }) => (
  <Container style={styles.container}>
    {props.navigation.openDrawer && (
      <Header>
        <Left>
          <Button transparent>
            <Icon active name="menu" onPress={() => props.navigation.openDrawer()} />
          </Button>
        </Left>
        <Body>
          <Title>Links</Title>
        </Body>
        <Right />
      </Header>
    )}
    <Content>
      <ScrollView style={styles.container}>
        {/* Go ahead and delete ExpoLinksView and replace it with your
         * content, we just wanted to provide you with some helpful links */}
        <Samples.ExpoLinksView />
      </ScrollView>
    </Content>
  </Container>
);
