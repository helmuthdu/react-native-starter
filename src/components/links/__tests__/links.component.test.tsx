import React from 'react';
import 'react-native';
import { Links } from '../links.component';
// Note: test renderer must be required after react-native.
import * as renderer from 'react-test-renderer';

const navigation = { navigate: jest.fn() } as any;

it('renders correctly', () => {
  const tree = renderer.create(<Links navigation={navigation} />).toJSON();
  expect(tree).toMatchSnapshot();
});
