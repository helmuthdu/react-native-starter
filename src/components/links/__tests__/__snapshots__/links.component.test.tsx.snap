// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
    ]
  }
/>
`;
